ziro-export-risport.ps1

```diff
@@ -0,0 +1,227 @@
+#Requires -Version 7.0
+
+class CucmConnector {
+   [string] $BaseUrl
+   [PSCredential] $Credential
+
+   CucmConnector(
+      [string] $BaseUrl,
+      [PSCredential] $Credential
+   ) {
+      $this.BaseUrl = $BaseUrl
+      $this.Credential = $Credential
+   }
+
+   [System.Xml.XmlDocument] ListPhone([int]$Skip, [int]$First) {
+      $uri = "{0}/axl/" -f $this.BaseUrl
+      $headers = @{
+         'Content-Type' = 'text/xml'
+         'SOAPAction'   = 'CUCM:DB ver=11.5 listPhone'
+      }
+
+      $requestBody = @"
+      <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns="http://www.cisco.com/AXL/API/11.5">
+         <soapenv:Header/>
+         <soapenv:Body>
+            <ns:listPhone>
+               <searchCriteria>
+                  <name>SEP%</name>
+               </searchCriteria>
+               <returnedTags>
+                  <name></name>
+                  <model></model>
+               </returnedTags>
+               <skip>${Skip}</skip>
+               <first>${First}</first>
+            </ns:listPhone>
+         </soapenv:Body>
+      </soapenv:Envelope>
+"@
+      
+      return Invoke-RestMethod -Method 'Post' -Authentication Basic -Credential $this.Credential -Headers $headers -Uri $uri -Body $requestBody -SkipCertificateCheck
+   }
+   
+   [System.Xml.XmlDocument] SelectCmDevice([string[]]$DeviceNames) {
+      $uri = "{0}/realtimeservice2/services/RISService70" -f $this.BaseUrl
+      $headers = @{
+         'Content-Type' = 'text/xml; charset=UTF-8'
+      }  
+
+      $bodySelectItemsList = $deviceNames | ForEach-Object { @"
+         <soap:item>
+            <soap:Item>${PSItem}</soap:Item>
+         </soap:item>
+"@}
+
+      $requestBody = @"
+      <soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:soap="http://schemas.cisco.com/ast/soap">
+         <soapenv:Header/>
+         <soapenv:Body>
+            <soap:selectCmDevice>
+               <soap:StateInfo></soap:StateInfo>
+               <soap:CmSelectionCriteria>
+                  <soap:MaxReturnedDevices>2000</soap:MaxReturnedDevices>
+                  <soap:DeviceClass>Phone</soap:DeviceClass>
+                  <soap:Model>255</soap:Model>
+                  <soap:Status>Any</soap:Status>
+                  <soap:NodeName></soap:NodeName>
+                  <soap:SelectBy>Name</soap:SelectBy>
+                  <soap:SelectItems>
+                     ${bodySelectItemsList}
+                  </soap:SelectItems>
+                  <soap:Protocol>Any</soap:Protocol>
+                  <soap:DownloadStatus>Any</soap:DownloadStatus>
+               </soap:CmSelectionCriteria>
+            </soap:selectCmDevice>
+         </soapenv:Body>
+      </soapenv:Envelope>
+"@
+      
+      $maxNumAttemptsForRateLimitError = 7
```

Comment:
I couldn't make sure the rate limit retry works because I was unable to trigger a rate limit error